storage/Utils/Logger.cc

```diff
@@ -89,18 +95,34 @@ namespace storage
     class LogfileLogger : public Logger
     {
     public:
+	LogfileLogger(const std::string& filename, int permissions = DEFAULT_PERMISSIONS);
 
 	virtual void write(LogLevel log_level, const std::string& component, const std::string& file,
 			   int line, const std::string& function, const std::string& content) override;
 
+    private:
+
+	static const int DEFAULT_PERMISSIONS = 0640;
+
+	const std::string filename;
+	const int permissions;
     };
 
 
+
+    LogfileLogger::LogfileLogger(const string& filename, int permissions) :
+	filename(filename),
+	permissions(permissions)
+    {
+    }
+
+
     void
     LogfileLogger::write(LogLevel log_level, const std::string& component, const std::string& file,
 			 int line, const std::string& function, const std::string& content)
     {
-	FILE* f = fopen("/var/log/libstorage-ng.log", "ae");
+	FILE* f = fdopen(open(filename.c_str(), O_WRONLY | O_APPEND | O_CREAT, permissions), "a");
```

Comment:
The "e" flag is missing.

Reply:
According to the documentation, "e" flag is ignored by fdopen.

Reply:
Thus the open call needs the flag. I would still add it to fdopen (for the developer).